File: src/main/scala/im/tox/toktok/app/call/CallOnGoingContactsAdapter.scala
package im.tox.toktok.app.call

import android.support.v7.widget.RecyclerView
import android.view.{LayoutInflater, ViewGroup}
import android.widget.{RelativeLayout, TextView}
import de.hdodenhof.circleimageview.CircleImageView
import im.tox.toktok.TR
import im.tox.toktok.TypedResource._
import im.tox.toktok.app.domain.Friend

final class CallOnGoingContactsAdapter(friends: java.util.List[Friend]) extends RecyclerView.Adapter[CallOnGoingContactsViewHolder] {

  def onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): CallOnGoingContactsViewHolder = {
    val itemView: RelativeLayout = LayoutInflater.from(viewGroup.getContext).inflate(TR.layout.call_ongoing_contact, viewGroup, false)
    new CallOnGoingContactsViewHolder(itemView)
  }

  def onBindViewHolder(viewHolder: CallOnGoingContactsViewHolder, position: Int): Unit = {
    val item: Friend = friends.get(position)
    viewHolder.mFriendName.setText(item.userName)
    viewHolder.mFriendImage.setImageResource(item.photoReference)
  }

  def getItemCount: Int = {
    friends.size()
  }

}

final class CallOnGoingContactsViewHolder(itemView: RelativeLayout) extends RecyclerView.ViewHolder(itemView) {
  val mFriendImage: CircleImageView = itemView.findView(TR.call_ongoing_contact_img)
  val mFriendName: TextView = itemView.findView(TR.call_ongoing_contact_name)
  val mFriendCallTime: TextView = itemView.findView(TR.call_ongoing_contact_calltime)
}

File: src/main/java/im/tox/toktok/app/call/CallOnGoingContactsAdapter.java
package im.tox.toktok.app.call;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;
import im.tox.toktok.R;
import im.tox.toktok.app.domain.Friend;

final class CallOnGoingContactsAdapter extends RecyclerView.Adapter<CallOnGoingContactsViewHolder> {

    private final List<Friend> friends;

    CallOnGoingContactsAdapter(List<Friend> friends) {
        this.friends = friends;
    }

    @Override
    public CallOnGoingContactsViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        RelativeLayout itemView = (RelativeLayout) LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.call_ongoing_contact, viewGroup, false);
        return new CallOnGoingContactsViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull CallOnGoingContactsViewHolder viewHolder, int position) {
        Friend item = friends.get(position);
        viewHolder.mFriendName.setText(item.userName);
        viewHolder.mFriendImage.setImageResource(item.photoReference);
    }

    @Override
    public int getItemCount() {
        return friends.size();
    }

}

final class CallOnGoingContactsViewHolder extends RecyclerView.ViewHolder {

    final CircleImageView mFriendImage;
    final TextView mFriendName;
    final TextView mFriendCallTime;

    CallOnGoingContactsViewHolder(@NonNull RelativeLayout itemView) {
        super(itemView);
        mFriendImage = itemView.findViewById(R.id.call_ongoing_contact_img);
        mFriendName = itemView.findViewById(R.id.call_ongoing_contact_name);
        mFriendCallTime = itemView.findViewById(R.id.call_ongoing_contact_call_time);
    }

}
